aws-predictions-tensorflow/src/main/java/com/amplifyframework/predictions/tensorflow/asset/TextClassificationDictionary.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.predictions.tensorflow.asset;
+
+import android.content.Context;
+import android.content.res.AssetManager;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.annotation.WorkerThread;
+
+import com.amplifyframework.core.Action;
+import com.amplifyframework.core.Consumer;
+import com.amplifyframework.predictions.PredictionsException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Loads the list of words and their numerical indices that the
+ * text classifier can recognize.
+ *
+ * The file being loaded should be in the format of:
+ * <pre>
+ * &#60;PAD&#62; 0
+ * &#60;START&#62; 1
+ * &#60;UNKNOWN&#62; 2
+ * word 3
+ * word 4
+ * ...
+ * </pre>
+ */
+public class TextClassificationDictionary implements Loadable<Map<String, Integer>, PredictionsException> {
+    private static final String DICTIONARY_PATH = "text_classification_vocab.txt";
+
+    // The maximum length of an input sentence.
+    private static final int MAX_SENTENCE_LENGTH = 256;
+
+    // Simple delimiter to split words.
+    private static final String DELIMITER_REGEX = "[ ,.!?\n]";
+
+    /*
+     * Reserved values in ImdbDataSet dictionary:
+     * dictionary["<PAD>"] = 0      used for padding
+     * dictionary["<START>"] = 1    mark for the start of a sentence
+     * dictionary["<UNKNOWN>"] = 2  mark for unknown words (OOV)
+     */
+    private static final String PAD = "<PAD>";
+    private static final String START = "<START>";
+    private static final String UNKNOWN = "<UNKNOWN>";
+
+    private final AssetManager assets;
+    private final Map<String, Integer> dictionary;
+
+    private Consumer<Map<String, Integer>> onLoaded;
+    private Consumer<PredictionsException> onError;
+    private Action onUnloaded;
+    private boolean loaded;
+
+    /**
+     * Constructs a loader for text classification dictionary.
+     * @param context the Android context
+     */
+    public TextClassificationDictionary(@NonNull Context context) {
```

Comment:
You can have this just take a dependency on Android's `Assets` directly, if that's all you need.

Reply:
I would prefer to leave it as is, since most Android developers should be more familiar with passing in an instance of `Context` than an `AssetManager`.